pyaedt/modeler/cad/Modeler.py

```diff
@@ -1134,6 +1207,388 @@ def _attributes(self):
         return coordinateSystemAttributes
 
 
+class ObjectCoordinateSystem(BaseCoordinateSystem, object):
+    """Manages object coordinate system data and execution.
+
+    Parameters
+    ----------
+    modeler :
+        Inherited parent object.
+    props : dict, optional
+        Dictionary of properties. The default is ``None``.
+    name : optional
+        The default is ``None``.
+    entity_id : int
+        Id of the entity object where the Object Coordinate System is anchored.
+
+    """
+
+    def __init__(self, modeler, props=None, name=None, entity_id=None):
+        BaseCoordinateSystem.__init__(self, modeler, name)
+        self.entity_id = entity_id
+        self._props = None
+        if props:
+            self._props = CsProps(self, props)
+            try:  # pragma: no cover
+                if "KernelVersion" in self.props:
+                    del self.props["KernelVersion"]
+            except:
+                pass
+        self._ref_cs = None
+
+    @property
+    def ref_cs(self):
+        """Reference coordinate system getter and setter.
+
+        Returns
+        -------
+        str
+        """
+        if self._ref_cs or settings.aedt_version <= "2022.2":
+            return self._ref_cs
+        obj1 = self._modeler.oeditor.GetChildObject(self.name)
+        self._ref_cs = obj1.GetPropValue("Reference CS")
+        return self._ref_cs
+
+    @ref_cs.setter
+    def ref_cs(self, value):
+        if settings.aedt_version <= "2022.2":
+            self._ref_cs = value
+            self.update()
+        obj1 = self._modeler.oeditor.GetChildObject(self.name)
+        try:
+            obj1.SetPropValue("Reference CS", value)
+            self._ref_cs = value
+        except:
+            self._modeler.logger.error("Failed to set Coordinate CS Reference.")
+
+    @property
+    def props(self):
+        """Properties of the coordinate system.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Modeler.CSProps`
+        """
+        if self._props or settings.aedt_version <= "2022.2" or self.name is None:
+            return self._props
+        self._get_coordinates_data()
+        return self._props
+
+    @property
+    def _part_name(self):
+        """Internally get the part name of the object where the coordinate system lays"""
```

Comment:
```suggestion
        """Internal part name of the object where the coordinate system lays."""
```
Properties should be described as nouns and not with a verb like "get". I'm not sure if "Internal" should be deleted from this description.